### ISSUE TITLE
Many new dart2js failures in Safari 9.1

### ISSUE THREAD

**Issue by whesse** [dart-lang/sdk#26114]
All our build slaves have updated to Mac OS X 10.11.4, with Safari 9.1, which is a big jump from safari 6 and 7 which were being tested before.

Many tests are failing, including core language tests, which don't use html or dom at all.

I have attached a list of failures from the bot.  Of 380 failures, 350 are in co19, 40 are in html, but 90 are in the language, corelib, and lib areas.  This is a major failure, and I am marking this issue high priority and severity.

The failures reproduce locally on a Mac laptop running OS X 10.11.4, with Safari 9.1.

I will not be suppressing these failures immediately in the status files, since I hope that we will be fixing most of them almost immediately.  I will keep updating this issue with lists of failures, and will also start breaking out the co19 failures to a separate issue, since we fix those with a lower priority.

[safari_9.1_errors.txt](https://github.com/dart-lang/sdk/files/193695/safari_9.1_errors.txt)

**Comment by hterkelsen** [comment]
From skimming https://chromegw.corp.google.com/i/client.dart/builders/dart2js-mac10.8-safari-be/builds/2987/steps/dart2js%20safari%20tests/logs/stdio it looks like the error patterns are not matching anything on Safari. For example, UnknownJsTypeError is thrown in dozens of tests, after all of the other type error patterns fail to match: https://github.com/dart-lang/sdk/blob/4fc394c96dea1749e3139161ac9a928a206473f6/sdk/lib/_internal/js_runtime/lib/js_helper.dart#L2259

**Comment by peter-ahe-google** [comment]
[CL 1842783002](https://codereview.chromium.org/1842783002/) fixes the type error matching in co19/Language/Expressions/Function_Invocation/Function_Expression_Invocation/static_type_t02 and possibly many more tests.

I'd appreciate if someone could land the CL on my behalf.

**Comment by hterkelsen** [comment]
Thanks Peter, it looks like your change fixed many of the errors. Now most of the test failures I see are either:

- tests that are expected to fail, but now pass
- tests that fail when calling 'obj instanceof ShadowRoot'

**Comment by hterkelsen** [comment]
Looks like most of the tests are fixed, here are the remaining failures from the safari bots:
```
FAILED: dart2js-safari release_ia32 html/js_test/JsArray Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/forms/datalist/datalist-child-validation_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/parsing-font-variant-ligatures_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/dom/css-selectorText_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/ex-unit-with-no-x-height_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/css-selector-text_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/pseudo-valid-unapplied_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/font-property-priority_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/inherited-properties-rare-text_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 co19/LayoutTests/fast/css/invalid-not-with-simple-selector-sequence_t01 Expected: Pass Actual: RuntimeError
FAILED: dart2js-safari release_ia32 html/audiobuffersourcenode_test/functional Expected: Pass Actual: RuntimeError
```

Looks like most of the remaining bugs have to do with CSS.

**Comment by alan-knight** [comment]
Thanks for compiling that. I'll take a look. I'm diving into co19 for
Jacob's Dartium changes today anyway.

**Comment by mit-mit** [comment]
@alan-knight what is the status here?

**Comment by whesse** [comment]
@peter-ahe-google has a CL that fixes the flaky failures on Safari.  I'm not sure if there are systematic failures that need to be fixed beyond that, and if they have separate issues filed.  Please answer, Peter Alan and Harry.  The CL is  https://codereview.chromium.org/1871883002/

**Comment by hterkelsen** [comment]
I think once that CL is in we can close this bug

**Comment by peter-ahe-google** [comment]
I agree, when I land my CL, this issue can be closed.

**Comment by kevmoo** [comment]
Looks like this is all about testing, right?

**Comment by whesse** [comment]
What about all the remaining non-flaky test failures?  Either this bug should be open until they all have bugs filed and their status updated, or there should be a new issue filed.  I think there are plenty of CSS-related and other tests whose status has not yet been updated.  I think nobody did anything with the list of failures from Harry, above.

**Comment by hterkelsen** [comment]
Here are the most recent failures (using Siggi's bot_failure_summary script):

```
$ bot_failure_summary dart2js-mac10.11-safari-1-3-be
Loading data from: http://build.chromium.org/p/client.dart/builders/dart2js-mac10.11-safari-1-3-be/builds/375/steps/steps/logs/stdio/text
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/framebuffer-test_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/object-deletion-behaviour_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/tex-image-and-sub-image-2d-with-canvas_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/tex-image-and-sub-image-2d-with-image-data-rgba4444_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/tex-image-and-sub-image-2d-with-image-data-rgba5551_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/tex-input-validation_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/tex-sub-image-2d_t01
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/dom/partial-layout-overlay-scrollbars_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/draw-webgl-to-canvas-2d_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/ex-unit-with-no-x-height_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/parsing-font-variant-ligatures_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/dom/HTMLLinkElement/resolve-url-on-insertion_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/dom/css-selectorText_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/dom/mutation-event-remove-inserted-node_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/forms/datalist/datalist-child-validation_t01
Pass => RuntimeError        dart2js-safari  co19 LibTest/html/Element/getBoundingClientRect_A01_t02
Pass => Timeout             dart2js-safari  html custom_elements_test/innerHtml
Pass => Timeout             dart2js-safari  html node_test/iterating
Total: 18 unexpected result(s), 8 now passing, 10 now failing.

$ bot_failure_summary dart2js-mac10.11-safari-2-3-be
Loading data from: http://build.chromium.org/p/client.dart/builders/dart2js-mac10.11-safari-2-3-be/builds/395/steps/steps/logs/stdio/text
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/uninitialized-test_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/css-selector-text_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/pseudo-valid-unapplied_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/flexbox/repaint-scrollbar_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/overflow/scrollbar-restored_t01
Pass => RuntimeError        dart2js-safari  co19 LibTest/html/Element/isContentEditable_A01_t01
Pass => RuntimeError        dart2js-safari  co19 LibTest/html/Element/isContentEditable_A02_t01
Pass => RuntimeError        dart2js-safari  co19 LibTest/html/IFrameElement/isContentEditable_A01_t01
Total: 8 unexpected result(s), 1 now passing, 7 now failing.

$ bot_failure_summary dart2js-mac10.11-safari-3-3-be
Loading data from: http://build.chromium.org/p/client.dart/builders/dart2js-mac10.11-safari-3-3-be/builds/409/steps/steps/logs/stdio/text
RuntimeError => Pass        dart2js-safari  co19 LayoutTests/fast/canvas/webgl/tex-image-and-sub-image-2d-with-image-data-rgb565_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/font-property-priority_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/inherited-properties-rare-text_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/css/invalid-not-with-simple-selector-sequence_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/dom/Range/remove-twice-crash_t01
Pass => RuntimeError        dart2js-safari  co19 LayoutTests/fast/events/scoped/editing-commands_t01
Pass => RuntimeError        dart2js-safari  html audiobuffersourcenode_test/functional
Fail => Timeout             dart2js-safari  html custom/attribute_changed_callback_test/unsupported_on_polyfill
Pass => Timeout             dart2js-safari  isolate static_function_test
Total: 9 unexpected result(s), 1 now passing, 8 now failing.
```

It looks like most of these are marked as failing in Peter's CL. I agree we should continue investigating the failures once Peter's CL lands.

**Comment by alan-knight** [comment]
Sorry, I said I'd look at those but have had too many high priority interrupts. However, I thought I did fix the status files for at least some of these.

In particular, I thought I'd fixed the suppressions for
 - things related to texImage and similar calls have probably been fixed by a couple of CL's I landed that  fix those.
 - There were strong mode issues related to the private class _MutationEvent, and it's strongly deprecated in DOM so we just removed it. So now if you get that event, which a couple of co19 tests do, it shows up as an opaque JS Object, not an Event. 

A couple were the issue we've seen with other browsers where they start printing things with quotes around them where they previously didn't, e.g. css-selector-text_t01. But I didn't try to fix those.

**Comment by whesse** [comment]
OK, great, I had forgotten there were status updates in Peter's CL, I was focused mainly on the code in it.  Then it should be fine to close the issue once it lands.

**Comment by mit-mit** [comment]
@peter-ahe-google is this on it's way in?

**Comment by peter-ahe-google** [comment]
Thank you for the reminder, I got lost in another project. I'll pick it up again on Monday.

**Comment by alan-knight** [comment]
Since Peter is looking at this, re-assigning.

**Comment by mit-mit** [comment]
This should not block 1.17 scheduled to release next week, so bumping to 1.18.

**Comment by peter-ahe-google** [comment]
I've attempted to land [CL 1871883002](https://codereview.chromium.org/1871883002) but it failed because it requires changes to files that aren't included in the Dart SDK. I'm unable to make further progress on this.

**Comment by mit-mit** [comment]
@peter-ahe-google looks like 1871883002 did land? Is there still an issue here?

**Comment by peter-ahe-google** [comment]
I'd appreciate if someone on the dart2js team could look in to this. My CL is landed, thanks to @whesse, and it should have removed most flakiness. However, the status files may still need some work.

**Comment by whesse** [comment]
The change is landed, all the flakiness is gone away, and the issue is resolved wonderfully.
